contrib/hamilton/contrib/user/zilto/lancedb_vdb/__init__.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from typing import Any, Dict, List
+
+logger = logging.getLogger(__name__)
+
+from hamilton import contrib
+
+with contrib.catch_import_errors(__name__, __file__, logger):
+    import pyarrow as pa
+    import lancedb
+
+from hamilton.function_modifiers import tag
+
+
+def vdb_client(uri: str = "./.lancedb") -> lancedb.DBConnection:
+    return lancedb.connect(uri=uri)
+
+
+@tag(side_effect="True")
+def table_ref(
+    vdb_client: lancedb.DBConnection,
+    table_name: str,
+    schema: pa.Schema,
+    overwrite_table: bool = False,
+) -> lancedb.db.LanceTable:
+    """Create or reference a LanceDB table
+
+    :param vdb_client: LanceDB connection.
+    :param table_name: Name of the table.
+    :param schema: Pyarrow schema defining the table schema.
+    :param overwrite_table: If True, overwrite existing table
+    :return: Reference to existing or newly created table.
+    """
+
+    try:
+        table = vdb_client.open_table(table_name)
+    except FileNotFoundError:
+        mode = "overwrite" if overwrite_table else "create"
+        table = vdb_client.create_table(name=table_name, schema=schema, mode=mode)
+
+    return table
+
+
+@tag(side_effect="True")
+def reset_vdb(vdb_client: lancedb.DBConnection) -> Dict[str, List[str]]:
+    """Drop all existing tables.
+
+    :param vdb_client: LanceDB connection.
+    :return: dictionary containing all the dropped tables.
+    """
+    tables_dropped = []
+    for table_name in vdb_client.table_names():
+        vdb_client.drop_table(table_name)
+        tables_dropped.append(table_name)
+
+    return dict(tables_dropped=tables_dropped)
+
+
+@tag(side_effect="True")
+def push_data(table_ref: lancedb.db.LanceTable, data: Any) -> Dict:
```

Comment:
@zilto oops -- merged too soon -- we should annotate it with `dict, list[dict], pd.DataFrame, or a Iterator[pa.RecordBatch]`